pkg/controller/job/job_controller_test.go

```diff
@@ -1853,14 +1853,33 @@ func TestSyncPastDeadlineJobFinished(t *testing.T) {
 	if !forget {
 		t.Errorf("Unexpected forget value. Expected %v, saw %v\n", true, forget)
 	}
-	if len(fakePodControl.Templates) != 0 {
-		t.Errorf("Unexpected number of creates.  Expected %d, saw %d\n", 0, len(fakePodControl.Templates))
+	wait.PollImmediate(1*time.Second, 5*time.Second, func() (done bool, err error) {
+		if manager.queue.Len() == 1 {
+			return true, nil
+		}
+		return false, nil
+	})
+	if manager.queue.Len() != 1 {
+		t.Errorf("Unexpected queue length. Expected %d, saw %d\n", 1, manager.queue.Len())
+	}
```

Comment:
I think that is better if you assert in the error returned by the helper
```suggestion
	err = wait.PollImmediate(1*time.Second, 5*time.Second, func() (done bool, err error) {
		if manager.queue.Len() == 1 {
			return true, nil
		}
		return false, nil
	})
	if err != nil {
		t.Errorf("Unexpected queue length. Expected %d, saw %d\n", 1, manager.queue.Len())
	}
```

anyway, I defer to current code owners for the review , thanks

Reply:
Sorry, I was doing that already. This was the code before I pushed the assertion of the error.